file-formats/sprv/type/zif_aff_sprv_v2.intf.abap

```diff
@@ -0,0 +1,41 @@
+INTERFACE zif_aff_sprv_v2
+  PUBLIC.
+
+
+  TYPES:
+    "! <p class="shorttext">Service Provider</p>
+    "! Service provider details
+    BEGIN OF ty_servprovider,
+      "! <p class="shorttext">Provider Name</p>
+      "! Name of the service provider
+      "! $required
+      provider_name      TYPE c LENGTH 30,
+      "! <p class="shorttext">Namespace</p>
+      "! Namespace of the service provider
+      "! $required
+      namespace          TYPE c LENGTH 255,
+      "! <p class="shorttext">Object Name Prefix</p>
+      "! Prefix for all generated objects
+      "! $showAlways
+      prefix             TYPE c LENGTH 10,
+      "! <p class="shorttext">Implementing Class</p>
+      "! Implementing Class of Service Provider
+      "! $showAlways
+      implementing_class TYPE c LENGTH 30,
```

Comment:
```suggestion
      implementing_class TYPE zif_aff_types_v1=>ty_object_name_30,
```
Minor: You can also use our predefined type for object names with 30 characters. Theoretically, you could also use the type for the provider_name type (this would also be compatible).